src/frontend/packages/suse-extensions/src/custom/kubernetes/services/kubernetes.analysis.service.ts

```diff
@@ -0,0 +1,305 @@
+import { KubeScoreReportHelper } from './kubescore-report.helper';
+import { PopeyeReportHelper } from './popeye-report.helper';
+import { environment } from 'frontend/packages/core/src/environments/environment';
+
+export interface KubernetesAnalysisType {
+  name: string;
+  id: string;
+  namespaceAware: boolean;
+  iconUrl?: string;
+  descriptionUrl?: string;
+}
+
+@Injectable()
+export class KubernetesAnalysisService {
+  kubeGuid: string;
+
+  public analyzers$: Observable<KubernetesAnalysisType[]>;
+  public namespaceAnalyzers$: Observable<KubernetesAnalysisType[]>;
+
+  public enabled$: Observable<boolean>;
+  public hideAnalysis$: Observable<boolean>;
+
+  constructor(
+    public kubeEndpointService: KubernetesEndpointService,
+    public activatedRoute: ActivatedRoute,
+    public store: Store<AppState>,
+    public http: HttpClient,
+    private snackBar: MatSnackBar,
+  ) {
+    this.kubeGuid = kubeEndpointService.kubeGuid;
+
+    // Is the backend plugin available?
+    this.enabled$ = this.store.select('auth').pipe(
+      map(auth => auth.sessionData.plugins && auth.sessionData.plugins.analysis)
+    );
+
+    this.hideAnalysis$ = this.enabled$.pipe(
+      startWith(true),
+      map(ok => !ok)
+    );
+
+    this.analyzers$ = of([
+      {
+        name: 'PopEye',
+        id: 'popeye',
+        namespaceAware: true,
+        iconUrl: '/core/assets/custom/popeye.png',
+        iconWidth: '80',
+        descriptionUrl: '/core/assets/custom/popeye.md'
+      },
+      {
+        name: 'Kube Score',
+        id: 'kube-score',
+        namespaceAware: true,
+        iconUrl: '/core/assets/custom/kubescore.png',
+        iconWidth: '120',
+        descriptionUrl: '/core/assets/custom/kubescore.md'
+      },
+      // {
+      //   name: 'Sonobuoy',
+      //   id: 'sonobuoy',
+      //   namespaceAware: false,
+      //   iconUrl: '/core/assets/custom/sonobuoy.png',
+      //   iconWidth: '70',
+      //   descriptionUrl: '/core/assets/custom/sonobuoy.md'
+      // }
+    ]);
+
+    this.namespaceAnalyzers$ = combineLatest(
+      this.analyzers$,
+      this.enabled$
+    ).pipe(
+      map(([a, enabled]) => {
+        if (!enabled) {
+          return null;
+        }
+        return a.filter(v => v.namespaceAware);
+      })
+    );
+  }
+
+    public delete(item) {
+    if (!Array.isArray(item)) {
+      item = [item];
+    }
+
+    const ids = [];
+    item.forEach(i => ids.push(i.id));
+
+    const proxyAPIVersion = environment.proxyAPIVersion;
+
+    // Fetch the report
+    const url = `/pp/${proxyAPIVersion}/analysis/reports`;
+    const headers = new HttpHeaders({});
+    const requestArgs = {
+      headers,
+      body: ids
+    };
+
+    const del = this.http.delete(url, requestArgs);
+    del.subscribe(d => {
+      const action = new GetAnalysisReports(this.kubeEndpointService.baseKube.guid);
+      this.store.dispatch(new ClearPaginationOfType(action));
```

Comment:
I _think_ the `ResetPagination` action should work better here, think that's the one that retains user filter/sort choices